versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5846 @@
+  }
+  
+  runtime {
+    container: "my_image:latest"
+  }
+}
+```
+
+If passed an array for the value of `bams`:
+
+|Element       |
+|--------------|
+|/path/to/1.bam|
+|/path/to/2.bam|
+|/path/to/3.bam|
+
+Would produce the command `python script.py --bams=/path/to/1.bam,/path/to/2.bam,/path/to/1.bam`
+
+#### Array serialization using write_lines()
+
+An array may be turned into a file with each element in the array occupying a line in the file.
+
+```wdl
+task test {
+  input {
+    Array[File] bams
+  }
+
+  command {
+    sh script.sh ${write_lines(bams)}
+  }
+  
+  runtime {
+    container: "my_image:latest"
+  }
+}
+```
+
+If `bams` is given this array:
+
+|Element       |
+|--------------|
+|/path/to/1.bam|
+|/path/to/2.bam|
+|/path/to/3.bam|
+
+Then, the resulting command line might be:
+
+```sh
+sh script.sh /jobs/564758/bams
+```
+
+Where `/jobs/564758/bams` would contain:
+
+```txt
+/path/to/1.bam
+/path/to/2.bam
+/path/to/3.bam
+```
+
+#### Array serialization using write_json()
+
+The array may be turned into a JSON document with the file path for the JSON file passed in as the parameter:
+
+```wdl
+task test {
+  input {
+    Array[File] bams
+  }
+
+  command {
+    sh script.sh ${write_json(bams)}
+  }
+  
+  runtime {
+    container: "my_image:latest"
+  }
+}
+```
+
+If `bams` is given this array:
+
+|Element       |
+|--------------|
+|/path/to/1.bam|
+|/path/to/2.bam|
+|/path/to/3.bam|
+
+Then, the resulting command line might look like:
+
+```sh
+sh script.sh /jobs/564758/bams.json
+```
+
+Where `/jobs/564758/bams.json` would contain:
+
+```json
+[
+  "/path/to/1.bam",
+  "/path/to/2.bam",
+  "/path/to/3.bam"
+]
+```
+
+#### Array deserialization using read_lines()
+
+`read_lines()` will return an `Array[String]` where each element in the array is a line in the file.
+
+This return value can be auto converted to other `Array` values.  For example:
+
+```wdl
+task test {
+  command <<<
+    python <<CODE
+    import random
+    for i in range(10):
+      print(random.randrange(10))
+    CODE
+  >>>
+
+  output {
+    Array[Int] my_ints = read_lines(stdout())
+  }
+  
+  runtime {
+    container: "my_image:latest"
+  }
+}
+```
+
+`my_ints` would contain ten random integers ranging from 0 to 10.
+
+#### Array deserialization using read_json()
+
+`read_json()` will return whatever data value resides in that JSON file.
+
+```wdl
+task test {
+  command <<<
+    echo '["foo", "bar"]'
+  >>>
+
+  output {
+    Array[String] my_array = read_json(stdout())
+  }
+  
+  runtime {
+    container: "my_image:latest"
+  }
+}
+```
+
+This task would assign the array with elements `"foo"` and `"bar"` to `my_array`.
+
+If the echo statement was instead `echo '{"foo": "bar"}'`, the engine MUST fail the task for a type mismatch.
+
+### Struct and Object
+
+`Struct`s and `Object`s are serialized identically. A JSON object is always deserialized to a WDL `Object`, which can then be coerced to a `struct` type if necessary.
```

Comment:
You don't provide any deserialization to a struct examples.  The section above and below both have a `write_json` and a `read_json` section.  Perhaps state why this section does not.    Also, consider removing Object here or mark with trash can for deprecation.

Reply:
Same explanation as above.